agent/api/task/task.go

```diff
@@ -3557,10 +3557,11 @@ func (task *Task) ToHostResources() map[string]*ecs.Resource {
 		}
 	} else {
 		containerMEMint64 := int64(0)
-		// To parse memory reservation / soft limit
-		hostConfig := &dockercontainer.HostConfig{}
 
 		for _, c := range task.Containers {
+			// To parse memory reservation / soft limit
+			hostConfig := &dockercontainer.HostConfig{}
+
```

Comment:
Can you update in description about how this is causing the problem? i.e. `json.Unmarshal` will not override previously assigned value if the field is not specified in the new json source.